extension/runtime/src/main/java/io/quarkiverse/temporal/WorkerFactoryRecorder.java

```diff
@@ -36,10 +38,14 @@ public WorkerFactoryRecorder(TemporalRuntimeConfig runtimeConfig, TemporalBuildt
     public Function<SyntheticCreationalContext<WorkerFactory>, WorkerFactory> createWorkerFactory(
             boolean isOpenTelemetryEnabled) {
         WorkerFactoryOptions.Builder options = WorkerFactoryOptions.newBuilder();
+        List<WorkerInterceptor> interceptors = CDI.current().select(WorkerInterceptor.class).stream()
+                .collect(Collectors.toList());
         if (isOpenTelemetryEnabled) {
-            options.setWorkerInterceptors(new OpenTracingWorkerInterceptor());
+            interceptors.add(new OpenTracingWorkerInterceptor());
         }
-        return context -> WorkerFactory.newInstance(context.getInjectedReference(WorkflowClient.class), options.build());
+        options.setWorkerInterceptors(interceptors.toArray(new WorkerInterceptor[0]));
+        return context -> WorkerFactory.newInstance(context.getInjectedReference(WorkflowClient.class),
```

Comment:
this is how I have been getting CDI dependencies in a synthetic creational context. 
It also need to be declared in the SyntheticBeanBuildItem.
Not sure why CDI.current() is working in this case

Reply:
but let's ask someone that is more qualified than me with synthetic bean to be sure !

Reply:
also I am wondering if the option creation should not be done inside the lambda

Reply:
OK i reverted those lambda changes. I won't be offended if you want to mess around with this PR to test the Synthetic changes or other CDI changes.

Reply:
Maybe @gsmet would know ? 

Is there a synthetic way of injecting all existing beans implementing a given interface without listing them explicitly in the synthetic injection points  (to get the same result as injecting Instance<Interceptor> ) ?

Reply:
here is the full change:

https://github.com/quarkiverse/quarkus-temporal/blob/fed9045c89905a63e9a29a3cae5cf65cd667912e/extension/runtime/src/main/java/io/quarkiverse/temporal/WorkerFactoryRecorder.java#L38-L49

Reply:
@rmanibus i could be wrong but the Synthetic is needed because WE created the `WorkflowClient` while the CDI.current() works because I am asking the running application if any downstream producers have created real beans?

Reply:
I am not sure, would prefer an external confirmation on this one !

Reply:
OK i asked on Zulip.

Reply:
From @mkouba 

> yes, it's better to use the synthetic injection point (i.e. BeanConfiguratorBase#addInjectionPoint() + SyntheticCreationalContext#getInjectedReference()) because this injection point is validated at build time and is also considered when removing unused beans...